### DISCUSSION TITLE
How can I see the React component associated to the element in the DOM?

### DISCUSSION THREAD

**Discussion by jpenna** [chakra-ui/chakra-ui#8667] [Q&A] [answered]
Hello all!

When using dev tools to see the DOM elements, there is no way to tell **at a glance** which element corresponds to which component. One has to hover over the element and see the selected element in the UI.

Is there any configuration to add comments with the component names on top of the elements or defined class names based on components?

Instead of having this:

![Screenshot from 2024-07-19 08-32-52](https://github.com/user-attachments/assets/4b69f0e3-c26c-4ac1-9d60-7e45ad27f17f)

It would be something like this:

![Screenshot from 2024-07-19 08-35-00](https://github.com/user-attachments/assets/33df7772-f592-4c1c-9152-cb8555a7f7fc)

**Comment by segunadebayo** [comment, marked as answer]
We are not injecting HTML comments or extra class names for DevTools.

Use React DevTools to map a node to a component. In v3, Zag also sets `data-scope` and `data-part` on the parts.
